ethereum/contracts/Bridge.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./Decoder.sol";
+import "./Application.sol";
+
+contract Bridge {
+    using Decoder for bytes;
+
+    uint64 public mostRecentBlock;
```

Comment:
To support pluggable verification modules, we want the verifier contract to handle replay protection. This allows the bridge to support any kind of verification in future. So I'd move this code to the verifier contract.

Reply:
Replaced event IDs and block numbers with just checking event hashes in https://github.com/Snowfork/polkadot-ethereum/pull/75/commits/0bf3adece367c08ce19bc50007eb975eb7de2e71. As it's a simple look-up map it's much easier to keep it in the Bridge contract for now. Given the several iterations on verification in the past few days, I'm hesitant to lock something in that we may end up changing as well. Can always move it to Verifier later on.